pkg/netceptor/netceptor.go

```diff
@@ -1656,6 +1656,86 @@ func (s *Netceptor) sendAndLogConnectionRejection(remoteNodeID string, ci *connI
 	return fmt.Errorf("rejected connection with node %s because %s", remoteNodeID, reason)
 }
 
+// PurgeDisconnectedNodes cleans up internal data of nodes that are no longer on mesh.
+func (s *Netceptor) PurgeDisconnectedNodes(target string) ([]string, error) {
+	s.knownNodeLock.Lock()
+	s.routingTableLock.Lock()
+	s.serviceAdsLock.Lock()
+	s.hashLock.Lock()
+	defer func() {
+		s.routingTableLock.Unlock()
+		s.knownNodeLock.Unlock()
+		s.serviceAdsLock.Unlock()
+		s.hashLock.Unlock()
+	}()
+
+	nodesWithRoutes := make(map[string]struct{})
+	for k, v := range s.routingTable {
+		nodesWithRoutes[k] = struct{}{}
+		nodesWithRoutes[v] = struct{}{}
+	}
+	nodesWithRoutes[s.nodeID] = struct{}{}
+
+	if target != "" {
+		if _, ok := nodesWithRoutes[target]; ok {
+			return nil, fmt.Errorf("target node is not disconnected")
```

Comment:
To be consistent with the params do we want to say "target unit_id is not disconnected"? If so, this is in a couple other strings below as well.

Reply:
calling the parameter `unit_id` was a mistake, it should have been `--node`. I've changed it